### DISCUSSION TITLE
Use deployless multicall as a fallback for chains where multicall3 is not deployed

### DISCUSSION THREAD

**Discussion by Gilgames000** [wevm/viem#183] [Idea / Feature Request] [answered]
I've been using [deployless multicall](https://destiner.io/blog/post/deployless-multicall/) as a fallback for the chains that my dapps support where multicall is not yet deployed. I'm using [this library](https://github.com/Destiner/ethcall) on top of ethers to achieve this. As I'm trying out viem and definitely plan to make the switch, it would be cool to have this same feature built-in over here since it already supports multicall out-of-the-box!

**Comment by jxom** [comment, marked as answer]
Agree! Would love to have this built in. Will put it on the Roadmap — this would be a nice first feature for someone to work on.

**Reply by not-reed** [reply]
opened a PR here: https://github.com/wagmi-dev/viem/pull/1443

**Comment by Gilgames000** [comment]
Just wanted to report that currently the deployless method which basically executes a read-only deployment of a contract using eth_call and overwrites its return value using assembly doesn't work with Anvil (it always worked with all of the other nodes I used in the past, Infura, Alchemy, Tenderly, Hardhat, etc.). I opened a discussion here foundry-rs/foundry/discussions/4549 for now.

**Comment by NBMSacha** [comment]
bump on this, would love to see this!
